File: dataStructures/linkedList/LinkedList.java
public class LinkedList {
    Node head;

    static class Node {
        int data;
        Node next;

        Node(int d) {
            data = d;
            next = null;
        }
    }

    public static LinkedList insert(LinkedList list, int data) {
        Node new_node = new Node(data);

        if (list.head == null) {
            list.head = new_node;
        } else {
            Node last = list.head;
            while (last.next != null) {
                last = last.next;
            }

            last.next = new_node;
        }

        return list;
    }

    public static void printList(LinkedList list) {
        Node currentNode = list.head;

        while (currentNode != null) {
            System.out.println(currentNode.data + ", ");
            currentNode = currentNode.next;
        }
    }

    public static void main(String[] args) {
        LinkedList list = new LinkedList();

        list = insert(list, 1);
        list = insert(list, 3);
        list = insert(list, 8);
        list = insert(list, 100);

        printList(list);
    }
}

File: python/dataStructures/linkedList/linked_list.py
class LinkedList:
    class Node:
        def __init__(self, value):
            self.value = value
            self.next = None
            
    def __init__(self):
        self.head = None # first node in the list

    def insert(self, data):
        new_node = self.Node(data)
        if self.head is None:
            self.head = new_node
            return
        current_node = self.head
        while current_node.next is not None:
            current_node = current_node.next
        current_node.next = new_node
    
    def remove(self):
        if self.head is None:
            print('Unable to remove empty linked list')
            return
        if self.head.next is None:
            self.head = None
            return
        self.head = self.head.next

    def printList(self):
        if self.head is None:
            print('Unable to print empty linked list')
            return
        current_node = self.head
        while current_node is not None:
            print(current_node.value)
            current_node = current_node.next

list = LinkedList()
list.insert(1)
list.insert(2)
list.insert(3)
list.insert(4)

list.printList()
